### PR TITLE
Split logic up which splits JWT

### PR DESCRIPTION
**Pull request by steadweb**
Added methods to get payload and header. Includes tests.

### CODE REVIEW

**Comment by steadweb** [comment]
I signed it!

File: composer.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
     },
     "autoload": {
         "psr-4": {
-            "Firebase\\JWT\\": "src"
+            "Firebase\\JWT\\": "src",
+            "Firebase\\JWT\\Tests\\": "tests"
```
Comment (bshaffer) [inline composer.json]: This is not necessary

File: src/JWT.php [hunk 2]
```diff
@@ -135,6 +131,61 @@ public static function decode($jwt, $key, $allowed_algs = array())
...
+    /**
+     * Splits the JWT into parts.
+     *
+     * @param string        $jwt            The JWT
+     *
+     * @throws UnexpectedValueException     Provided JWT was invalid
+     *
+     * @return array
+     */
+    public static function split($jwt)
```
Comment (bshaffer) [inline src/JWT.php]: This should be a private function

File: tests/HelperTest.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Firebase\JWT\Tests;
+
+use \Firebase\JWT\JWT;
+use \UnexpectedValueException;
```
Comment (bshaffer) [inline tests/HelperTest.php]: unnecessary leading slashes in import statements

File: src/JWT.php [hunk 4]
```diff
@@ -74,19 +74,15 @@ public static function decode($jwt, $key, $allowed_algs = array())
...
-        $tks = explode('.', $jwt);
-        if (count($tks) != 3) {
-            throw new UnexpectedValueException('Wrong number of segments');
-        }
-        list($headb64, $bodyb64, $cryptob64) = $tks;
+        list($headb64, $bodyb64, $cryptob64) = static::split($jwt);
         if (null === ($header = static::jsonDecode(static::urlsafeB64Decode($headb64)))) {
             throw new UnexpectedValueException('Invalid header encoding');
         }
         if (null === $payload = static::jsonDecode(static::urlsafeB64Decode($bodyb64))) {
```
Comment (bshaffer) [inline src/JWT.php]: This logic is now duplicated here and also in the `header()/payload()` functions

**Comment by steadweb** [comment]
@bshaffer thanks for the reply! 

My aim was to sniff the `$headers` before decoding, as if the signed `JWT` is invalid, it'll throw an exception, which is accetable. But I want to check the `aud` or indeed the app that signed the `JWT`, before I decode it.

I'm working on a concept where I store the public key of the JWT within a SaaS app and that's registered by other apps. These are defined by their domain, which should be passed within the `JWT` header.

**Comment by bshaffer** [comment]
In that case, I would suggest using the methods here to extract the headers manually:

```php
use Firebase\JWT;
// your application code
list($headb64, $bodyb64, $cryptob64) = explode('.', $jwt);
$headers = JWT::jsonDecode(JWT::urlsafeB64Decode($headb64));
```

**Pull request closed without merging**